superqt/combobox/_enum_combobox.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional, Type, TypeVar, Union
+from enum import Enum
+
+from ..qtcompat.QtWidgets import QComboBox
+from ..qtcompat.QtCore import Signal, PYQT5
+
+EnumType = TypeVar("EnumType", bound=Enum)
+enum_type = Enum if PYQT5 else object
+
+def _get_name(enum_value: Enum):
+    """Create human readable name if user does not provide own implementation of __str__"""
+    if enum_value.__str__.__module__ != "enum":
+        # check if function was overloaded
+        name = str(enum_value)
+    else:
+        name = enum_value.name.replace("_", " ")
+    return name
+
+
+class EnumComboBox(QComboBox):
+    """
+    This is implementation of variant of EnumComboBox for select enum value.
+    If enum class does not provide own __str__ implementation then human readable name is provide using name of item
+    replacing underscores with spaces.
+    """
+    currentValueChanged = Signal(enum_type)
+
+    def __init__(self, parent=None, enum:Optional[Type[EnumType]] = None):
+        super().__init__(parent)
+        self._enum = enum
+        if enum is not None:
+            self.setEnum(enum)
+        self.currentIndexChanged.connect(self._emit_signal)
+
+    def setEnum(self, enum: Type[EnumType]):
+        """
+        Set enum class from which members value should be selected
+        """
+        self.clear()
+        self._enum = enum
+        super().addItems(list(map(_get_name, enum.__members__.values())))
+
+    def enum(self) -> Optional[Type[EnumType]]:
+        return self._enum
+
+    def clear(self):
+        self._enum = None
+        super().clear()
+
+    def currentValue(self) -> EnumType:
```

Comment:
curious whether you considered and rejected using `currentData`?  Do you think it's best to leave that for user usage?

I'm just thinking about alternatives to "value" ... it doesn't immediately imply to me that we're getting the enum member...

Reply:
I don't think about `currentData`, I miss it.
But `currentData` should return QVariant so return Enum from `currentData` may break Qt API compatibility?

Reply:
in my experience qvariant has worked pretty well with most python types... however, after playing with it a bit, I think I've changed my mind: I think we should leave `currentData` for user-space.

I still don't love `currentValue` though... so what do you think about the following?

`setEnumClass` -> clear combo and set enum class (replaces current `setEnum`)
`enumClass` -> return the enum class  (replaces current `enum`)

`currentEnum` -> returns the currently selected enum _member_
`setCurrentEnum` -> set the currently selected enum _member_ (and mark as a slot to match [slots](https://doc.qt.io/qt-5/qcombobox.html#public-slots))
change `currentValueChanged` to `currentEnumChanged`

(I'm still thinking this through, so let me know if you don't like it)

Reply:
changed. I`m not sure if `setCurrentEnum` or `setEnum` should be used.

Reply:
qt seems to use `setCurrentText` and `setCurrentIndex` (instead of `setText` and `setIndex`) ... so that's why I suggested that pattern.